boot/bootutil/src/loader.c

```diff
@@ -107,7 +107,7 @@ boot_read_image_headers(struct boot_loader_state *state, bool require_all,
              *
              * Failure to read any headers is a fatal error.
              */
-            if (i > 0 && !require_all) {
+            if (rc < 0 && !require_all) {
```

Comment:
Could you please explain why this change is required?
I understand the need to check for a negative `rc`, but considering the above comment, the check `i > 0` seems to be required.

Reply:
With i >0 an error is return if primary slot has no valid header, it prevents from parsing the secondary slot header, and installing an image on a secondary slot. Your remark is valid, and I have  another fix that I should update soon, there is more changes... Let's see automatic test result and then the review comment.